process/physics.py

```diff
@@ -379,389 +498,966 @@ def conhas(alphaj, alphap, bt, delta95, eps, kappa95, p0, rmu0):
+        )
+        + (1.2 / (1.0 + 0.5 * charge_profile) * f32ee_teff**4)
     )
 
     # $F_{32\_ei}(Y)$, Eq.15c
-    hcei = (
-        (-(0.56 + 1.93 * zz) / zz / (1.0 + 0.44 * zz) * (zfti - zfti4))
-        + (4.95 / (1.0 + 2.48 * zz) * (zfti2 - zfti4 - 0.55 * (zfti3 - zfti4)))
-        - (1.2 / (1.0 + 0.5 * zz) * zfti4)
+    big_f32ei_teff = (
+        (
+            -(0.56 + 1.93 * charge_profile)
+            / charge_profile
+            / (1.0 + 0.44 * charge_profile)
+            * (f32ei_teff - f32ei_teff**4)
+        )
+        + (
+            4.95
+            / (1.0 + 2.48 * charge_profile)
+            * (
+                f32ei_teff**2
+                - f32ei_teff**4
+                - 0.55 * (f32ei_teff**3 - f32ei_teff**4)
+            )
+        )
+        - (1.2 / (1.0 + 0.5 * charge_profile) * f32ei_teff**4)
     )
 
+    # big_f32ee_teff + big_f32ei_teff = L32 coefficient
+
     # Corrections suggested by Fable, 15/05/2015
-    return beta_poloidal_local(j, nr, rmajor, bt, ne, tempe, mu, rho) * (
-        hcee + hcei
-    ) + dcsa(
-        j, nr, rmajor, bt, triang, ne, ni, tempe, tempi, mu, rho, zef, sqeps
-    ) * beta_poloidal_local(
-        j, nr, rmajor, bt, ne, tempe, mu, rho
-    ) / beta_poloidal_local_total(
-        j, nr, rmajor, bt, ne, ni, tempe, tempi, mu, rho
+    return beta_poloidal_sauter(
+        radial_elements, number_of_elements, rmajor, bt, ne, tempe, inverse_q, rho
+    ) * (big_f32ee_teff + big_f32ei_teff) + calculate_l31_coefficient(
+        radial_elements,
+        number_of_elements,
+        rmajor,
+        bt,
+        triang,
+        ne,
+        ni,
+        tempe,
+        tempi,
+        inverse_q,
+        rho,
+        zeff,
+        sqeps,
+    ) * beta_poloidal_sauter(
+        radial_elements, number_of_elements, rmajor, bt, ne, tempe, inverse_q, rho
+    ) / beta_poloidal_total_sauter(
+        radial_elements,
+        number_of_elements,
+        rmajor,
+        bt,
+        ne,
+        ni,
+        tempe,
+        tempi,
+        inverse_q,
+        rho,
     )
 
 
 @nb.jit(nopython=True, cache=True)
-def xcsa(
-    j,
-    nr,
-    rmajor,
-    bt,
-    triang,
-    mu,
-    sqeps,
-    tempi,
-    tempe,
-    amain,
-    zmain,
-    ni,
-    ne,
-    rho,
-    zef,
-):
-    """Grad(ln(Ti)) coefficient in the Sauter bootstrap scaling
-    author: P J Knight, CCFE, Culham Science Centre
-    j  : input integer : radial element index in range 2 to nr
-    nr : input integer : maximum value of j
-    This function calculates the coefficient scaling grad(ln(Ti))
-    in the Sauter bootstrap current scaling.
-    Code by Angioni, 29th May 2002.
-    <P>The code was supplied by Emiliano Fable, IPP Garching
-    (private communication).
-    O. Sauter, C. Angioni and Y. R. Lin-Liu,
-    Physics of Plasmas <B>6</B> (1999) 2834
-    O. Sauter, C. Angioni and Y. R. Lin-Liu, (ERRATA)
-    Physics of Plasmas <B>9</B> (2002) 5140
+def calculate_l34_alpha_31_coefficient(
```

Comment:
Should this be an internal function (ie named with an _)?

Reply:
Yes, changed